### ISSUE TITLE
Rerendering of inner part of window on maximize/restore

### ISSUE THREAD

**Issue by mgrouch** [rjaros/kvision#374]
I see in desktop example TextEditor, Paint, WebBrowser override height setters. 
It’s only working in example. In more generic case I set window size on maximize using 
calc(100vh - XXXpx). 

How is it possible to achieve inner part to size automatically to fill maximized/restored  window in generic way?

thanks

**Comment by rjaros** [comment]
It just so happened that the desktop example uses windows that contain specific html elements - iframe, canvas and a custom rich text editor. These elements just don't scale well automatically. That's why they are scaled manually by overriding setters and event handlers. In general it's not required at all, because you can easily use relative sizing (e.g. with percent units).
See this example:
```kotlin
class Win : Window("Test window", 300.px, 300.px, isDraggable = true, isResizable = true, maximizeButton = true) {

    var isMaximized = false
    var storedWidth: CssSize? = null
    var storedHeight: CssSize? = null
    var storedTop: CssSize? = null
    var storedLeft: CssSize? = null
    init {
        simplePanel {  // This panel will automatically fit the window size
            width = 100.perc
            height = 100.perc
            background = Background(Color.name(Col.SILVER))
        }
    }

    override fun toggleMaximize() {
        if (isMaximized) {
            width = storedWidth
            height = storedHeight
            top = storedTop
            left = storedLeft
            isMaximized = false
        } else {
            storedWidth = width
            storedHeight = height
            storedTop = top
            storedLeft = left
            top = 100.px
            left = 0.px
            setStyle("height", "calc(100vh - 100px)")
            width = 100.vw
            isMaximized = true
        }
    }
}
```

**Comment by mgrouch** [comment]
Those 3 TextEditor, Paint, WebBrowser are giving me hard time. They do not resize in my tests on window maximizeToggle after second time of maximize/restore cycle . First time is fine. They even ignore height = 
completely

**Comment by rjaros** [comment]
Could you please share your code?

**Comment by mgrouch** [comment]
I'll try to get to improving desktop example to make it more generic.
I think some of the classes related to window management can be made common and be a part of the framework,
after taking some time looking at how most of the people would use them.
Awesome framework and fun to code using it. Unfortunately sometimes I had to dig into HTML and CSS.
Would be nice to reach level when coding UI is getting close to the way someone would code native UI app.
Hot module replace is so much better than (venerable) GWT.

**Comment by mgrouch** [comment]
Looks like minimizing a window removes resize listener on other windows

**Comment by rjaros** [comment]
The problem was caused by html elements clash (snabbdom is reusing the same html elements for different components when it's possible). It will be fixed in the next release.

**Comment by rjaros** [comment]
Fixed in 5.10.0
